sphinx/builders/linkcheck.py

```diff
@@ -134,6 +136,10 @@ def check_uri():
             # split off anchor
             if '#' in uri:
                 req_url, anchor = uri.split('#', 1)
+                for rex in self.anchors_ignore:
+                    if rex.match(uri):
```

Comment:
My thought was that if the same anchor was used on multiple sites, that it would be quite hard to exclude just that anchor if it was the only one problematic, without also matching the other sites:

- http://sphinx-doc.com/#top
- http://example.com/#top

However probably can start with a simple match, and then add an option to enable full URI comparison if someone encounters such a scenario.